services/api/analyzers.js

```diff
@@ -265,16 +272,88 @@ function deleteAnalyzer(req, res, next) {
                 return next(createError(500, reply['error']['message']))
             }
             closeResponse()
-            models['analyzers'].findByIdAndRemove(id, (err) => {
-                if (err) {
-                    return next(createError(500, null, err))
-                }
-                res.status(204).send()
+
+            // get all corresponding events
+            models['events'].find({analyzerId: id})
+            .then((events) => {
+                // collect & delete all objects key of all events id
+                objKeys = []
+                forEach(events, (event) => {
+                    forEach(event.content, (value) => {
+                        // Each type of events has its own content structure. To
+                        // generalize, we assume object key is stored in string type.
+                        if (isString(value)) {
+                            objKeys.push(value);
+                        }
+                    });
+                });
+                // Delete all objects by keys.
+                return deleteObjects(objKeys).then(() => {return events});
+            })
+            .then((events) => {
+                // delete all corresponding events
+                const eventIds = map(events, event => event._id);
+                return models['events'].deleteMany({
+                    _id: {
+                        $in: eventIds,
+                    },
+                });
+            })
+            .then(() => {
+                // delete analyzer
+                return models['analyzers'].findByIdAndRemove(id);
+            })
+            .then(() => {
+                return res.status(204).send();
             })
+            .catch((err) => {
+                return next(createError(500, null, err))
+            });
+
         })
     })
 }
 
+function deleteObjects(keys) {
+    return new Promise((resolve, reject) => {
+        // Get configurations.
+        const { obj_storage: objectStorageConfig } = config.services;
+        const {
+            endpoint_url: endpoint,
+            bucket_name: bucketName,
+        } = objectStorageConfig.params;
+        const {
+            access_key: accessKeyId,
+            secret_key: secretAccessKey,
+        } = objectStorageConfig.credentials;
+
+        // Connect to object store.
+        const store = new S3({
```

Comment:
We could create a global object-store client so that we don't have to connect to object store every time we call this function.